src/main/webapp/app/shared/competency-selection/competency-selection.component.ts

```diff
@@ -40,6 +40,12 @@ export class CompetencySelectionComponent implements OnInit, ControlValueAccesso
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     _onChange = (value: any) => {};
 
+    protected readonly highLinkWeight = 1;
+    protected readonly mediumLinkWeight = 0.5;
+    protected readonly lowLinkWeight = 0.25;
+    protected readonly lowLinkWeightCutOff = 0.375; // halfway between low and medium
+    protected readonly mediumLinkWeightCutOff = 0.75; // halfway between medium and high
```

Comment:
We had defined constants to be screaming snake case be default, as this is a widespread convention https://github.com/ls1intum/Artemis/pull/9564#pullrequestreview-2392135148